src/main/java/com/teamscale/upload/TeamscaleUpload.java

```diff
@@ -59,350 +35,53 @@ public static void main(String[] args) throws Exception {
 			LogUtils.enableStackTracePrintingForKnownErrors();
 		}
 
-		Map<String, Set<File>> formatToFiles = ReportPatternUtils.resolveInputFilePatterns(commandLine.inputFile,
-				commandLine.files, commandLine.format);
-
-		try {
-			convertXCodeReports(formatToFiles);
-		} catch (ConversionException e) {
-			LogUtils.failWithoutStackTrace(e.getMessage(), e);
-		}
-
-		OkHttpClient client = OkHttpUtils.createClient(commandLine.validateSsl, commandLine.getKeyStorePath(),
-				commandLine.getKeyStorePassword(), commandLine.getTimeoutInSeconds());
-		try {
-			performUpload(client, formatToFiles, commandLine);
-		} catch (SSLHandshakeException e) {
-			handleSslConnectionFailure(commandLine, e);
-		} finally {
-			// we must shut down OkHttp as otherwise it will leave threads running and
-			// prevent JVM shutdown
-			client.dispatcher().executorService().shutdownNow();
-			client.connectionPool().evictAll();
-		}
+		Map<String, Set<File>> filesByFormat = resolveAndConvertFiles(commandLine);
+		TeamscaleClient.performUpload(commandLine, filesByFormat);
 	}
 
 	/**
-	 * Converts the reports from the internal binary XCode format to a readable
-	 * report that can be uploaded to Teamscale.
+	 * Resolves the files that should be uploaded to Teamscale and converts them to
+	 * the expected formated if needed (e.g., XCode reports).
 	 */
-	private static void convertXCodeReports(Map<String, Set<File>> formatToFiles) throws ConversionException {
```

Comment:
For the reviewer: I felt like this code does not belong here. Hence I moved it to the `XCResultConverter`